lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -797,7 +798,6 @@ def configure_network(self, vm_obj):
                 dvs_port_connection.switchUuid = pg_obj.config.distributedVirtualSwitch.uuid
                 nic.device.backing = vim.vm.device.VirtualEthernetCard.DistributedVirtualPortBackingInfo()
                 nic.device.backing.port = dvs_port_connection
-                nic_change_detected = True
```

Comment:
Any specific reason to remove this ?

Reply:
If changes are detected nic_change_detected is set to true line 791 or 794.
This line break ansible idempotency because nic_change_detected is always set to true even if there is no changes on networking.